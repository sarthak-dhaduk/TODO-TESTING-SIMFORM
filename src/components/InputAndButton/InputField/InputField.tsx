import React, { useEffect, useState } from "react";
import { InputFieldProps } from "./InputField.types";

function InputField({ optionHandler, addNewTask }: InputFieldProps) {
  const [value, setValue] = useState("");

  //on submit adding task and clearing the value of inputField
  const formSubmitHandler = (e: React.KeyboardEvent<HTMLFormElement>) => {
    e.preventDefault();
    addNewTask(value);
    setValue("");
  };

  //on escape keyPress toggling the input and btn
  useEffect(() => {
    function keyPressHandler(e: KeyboardEvent) {
      if (e.key === "Escape") {
        optionHandler();
      }
    }
    document.addEventListener("keydown", keyPressHandler);

    return () => document.removeEventListener("keydown", keyPressHandler);
  }, [optionHandler]);

  return (
    <form onSubmit={formSubmitHandler}>
      <input
        className="w-full my-4 text-textLight bg-white focus:outline-0"
        placeholder="Enter the Task !"
        value={value}
        onChange={e => setValue(e.target.value)}
        autoFocus
      />
    </form>
  );
}

export default InputField;
