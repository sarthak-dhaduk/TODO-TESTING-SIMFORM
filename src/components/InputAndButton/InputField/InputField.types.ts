export type InputFieldProps = {
  optionHandler: () => void;
  addNewTask: (value: string) => void;
};
