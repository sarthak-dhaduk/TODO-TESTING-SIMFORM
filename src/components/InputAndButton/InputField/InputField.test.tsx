import { render, renderHook, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import user from "@testing-library/user-event";
import InputField from "./InputField";

describe("InputField", () => {
  //testing the ui snapshot
  test("InputField renders correctly", () => {
    const renderedComponent = renderer
      .create(<InputField optionHandler={() => {}} addNewTask={() => {}} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  //events with btn are handled
  test("should add and remove event listener on mount and unmount", () => {
    const addEventListenerMock = jest.spyOn(document, "addEventListener");
    const removeEventListenerMock = jest.spyOn(document, "removeEventListener");

    const { unmount } = render(
      <InputField optionHandler={() => {}} addNewTask={() => {}} />
    );

    expect(addEventListenerMock).toHaveBeenCalledWith(
      "keydown",
      expect.any(Function)
    );

    unmount();

    expect(removeEventListenerMock).toHaveBeenCalledWith(
      "keydown",
      expect.any(Function)
    );
  });
});
