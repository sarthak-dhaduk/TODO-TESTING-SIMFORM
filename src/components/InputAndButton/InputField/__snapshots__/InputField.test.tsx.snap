// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField InputField renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    className="w-full my-4 text-textLight bg-white focus:outline-0"
    onChange={[Function]}
    placeholder="Enter the Task !"
    value=""
  />
</form>
`;
